Code cleanup, optimisations and lint

vernig opened the pull request:
<!-- Describe your Pull Request -->

The aim of this PR is to refactor the solution backend code (Twilio Serverless) and clean it up of unused functions, routines and variables. 

The main additions in this PR are:

* public/resources/functions/helpers.private.js: this file contains helpers functions used by the other endpoints
* public/resources/functions/options.private.js: this file contains a single object with all the options used by the solution backed

This PR also:
Fix #18
Fix #21


> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

vernig commented:
@ktalebian I rebased my PR on #24. 

I disabled some of the `eslint` in the serverless, namely: 
* `camelcase`: this requires change in the plugin. We can do that in a follow-up PR
* `sonarjs/cognitive-complexity`: this needs a bigger refactoring (and more time)

We need more end-to-end testing before merging (cc @bermudezmt )

vernig commented:
@ktalebian @bermudezmt I've done some end-to-end testing and fix some bugs. The code is ready for review / testing.

ktalebian commented on public/resources/functions/helpers.private.js:
```diff
@@ -0,0 +1,94 @@
+  return fetchTask
+    .then((result) => {
+      const task = Array.isArray(result) ? result[0] : result;
+      res = {
```
Redundant - `return { ... }`

vernig replied:
Solved in e98d19b

ktalebian commented on public/resources/functions/helpers.private.js:
```diff
@@ -0,0 +1,94 @@
+      return res;
+    })
+    .catch((error) => {
+      res = {
```
Same here

vernig replied:
Solved in e98d19b

ktalebian commented on public/resources/functions/helpers.private.js:
```diff
@@ -0,0 +1,94 @@
+  const now = new Date();
+  const timeRecvd = moment(now);
+  return {
+    time_recvd: timeRecvd,
```
Can you convert these two `camelCase`?

vernig replied:
I'll do it in a follow-up PR @ktalebian because they require changes in the plugin code as well. I want to keep this PR only for serverless

vernig replied:
Raised #28 to track this

ktalebian commented on public/resources/functions/helpers.private.js:
```diff
@@ -0,0 +1,94 @@
+    });
+}
+
+async function cancelTask(client, workspaceSid, taskSid) {
```
Would be nice to have a js docblock for these methods as well!

vernig replied:
Solved in e98d19b

ktalebian commented on public/resources/functions/inqueue-utils.js:
```diff
@@ -157,104 +143,85 @@ exports.handler = JWEValidator(async function (context, event, callback) {
-        callback(null, resp.setBody(res));
-      }
-      deleteMain();
+      let result = await getTask(event.taskSid);
```
better to call these different variables? 

```js
const task = await getTask(event.taskSid);
const updatedTask = await updateTask(event.taskSid, task.attr);
```

vernig replied:
Solved in e98d19b

ktalebian commented on public/resources/functions/inqueue-utils.js:
```diff
@@ -157,104 +143,85 @@ exports.handler = JWEValidator(async function (context, event, callback) {
+       * original task in Twilio WFO
+       */
+      if (!newAttributes.hasOwnProperty('conversations')) {
+        newAttributes = Object.assign(newAttributes, {
```
Cleaner to use spread operator:

```js
newAttributes = { ...newAttributes, { conversations: { conversation_id: event.taskSid } } };
```

vernig replied:
Solved in e98d19b

bermudezmt reviewed (changes requested):
Tested the queued callback and all was working as expected. Testing voicemail however led to a Function execution error (Error - 82005):

```
Function execution resulted in an error log: ETIMEDOUT | stack: Error: ETIMEDOUT
    at Timeout.<anonymous> (/var/task/node_modules/request/request.js:848:19)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
 ```
 @vernig Since I added you to that account, maybe you can review the logs in the debugger.

vernig commented:
Thanks again @bermudezmt for testing the code. I actually found a bug (introduced when I cleaned up the code). It's now fixed in d68ed62. Thanks again for spotting it!

bermudezmt reviewed (approved):
LGTM after the follow-up commits. @ktalebian feel free to merge if you're satisfied with the changes.

ktalebian commented on public/resources/functions/helpers.private.js:
```diff
@@ -0,0 +1,115 @@
+ * @param {string} timeZone Timezone name
+ * @returns {Object}
+ */
+//
```
Nitpick: remove this

vernig replied:
Solved in ab52287

vernig commented:
@rnairtwilio do you want to have a look to the PR before I merge it?
